16_bg_slider/app.js

```diff
@@ -0,0 +1,34 @@
+const btnPrev = document.querySelector(".prev")
+const btnNext = document.querySelector(".next")
+
+var slideIndex = 1;
+showSlides(slideIndex);
+
+function nextImage(n) {
+    showSlides(slideIndex += n)
+}
+
+function currentSlide(n) {
+    showSlides(slideIndex = n);
+}
+
+function showSlides(n) {
+    var i;
+    var slides = document.getElementsByClassName('slides');
+    if (n > slides.length) {
+        slideIndex = 1
+    }
+    if (n < 1) {
+        slideIndex = slides.length
+    }
+    for (i = 0; i < slides.length; i++) {
+        slides[i].style.display = "none"
+    }
+    slides[slideIndex - 1].style.display = "block";
```

Comment:
You mentioned this fails. Here's a hint: Do a of `console.log(slides[slideIndex -1])` before this line and see its output.

There's a powerful devTools feature that helps you to debug code right in the browser. Instead of `console.log()` write `debugger;`. The code will stop when reached, like if you "stop in the time" and you can see what's happening in that precise moment. Here's a [short video about `debugger;`](https://www.youtube.com/watch?v=ABlaMXkUwzY).

Good luck with this challenge, I'm looking forward to see your final solution! :D